device.c

```diff
@@ -133,6 +149,14 @@ static int vcam_try_fmt_vid_cap(struct file *file,
         f->fmt.pix.height = dev->output_format.height;
     }
 
+    if (dev->conv_res_on) {
+        const struct v4l2_frmsize_discrete *sz = v4l2_find_nearest_size(
+            vcam_sizes, ARRAY_SIZE(vcam_sizes), width, height, 1280, 720);
```

Comment:
You should explain and address the operation to set 720p.

Reply:
`v4l2_find_nearest_size` check the nearest size of `1280, 720` from the array of `vcam_sizes`, and it assign the nearest size to `sz`.
After  `sz` gets the size, it assign to `f->fmt.pix.height` and `width`.
`vcam_update_format_cap` reset the `vcam_in_queue` from the `640 * 480` to `1280 * 720`.

Reply:
The value pair `(1280, 720)` is already set earlier, and you can avoid its appearance here. The intention was to iterate the each supported resolution for converter.

Reply:
Actually, the parameters `(1280, 720)` in `v4l2_find_nearest_size` can be changed to below `(1280, 720)`.  
For example,  you input `(600, 300)` and vcam will convert to `(640, 360)` because I define  `(640, 360)`  in `vcam_sizes` array.
If you input `(600, 400)`, vcam will convert to `(640,480)`.

By the way,  I will include an header which defined `image size` in v4l2. To avoid hardcode `(1280, 720)` in here.